Sources/Noora/Components/TextPrompt.swift

```diff
@@ -36,7 +41,14 @@ struct TextPrompt {
 
         logger?.debug("Prompted '\(prompt.plain())'")
 
-        render(input: input, withCursor: false)
+        let validationResult = Validator().validate(input: input, rules: validationRules)
```

Comment:
I'd consider either:
- Keeping the validate method as a private method
- Passing the instance through `init`, so we respect our dependency injection convention. We can keep the default value at all times for now.

Reply:
@fortmarek, thanks for your comment!

For this particular case, eliminating the `Validator` object seems reasonable, along with making the `validate` method private. However, on the other hand, the `TextPrompt` object, as I see it, would take on unnecessary responsibility for validation. I believe it's better to keep the `Validator` object, which is responsible for handling validation. 

Passing the instance through `init` is a good approach, but in this case, it seems we would need to make the `Validator` object public, which doesn't seem ideal. 

What do you think about this?

Reply:
> Passing the instance through init is a good approach, but in this case, it seems we would need to make the Validator object public, which doesn't seem ideal.

You don't have to make it public by providing an additional internal init along with a public one. It does mean some extra boilerplate, but I would like to make the fact that we're depending on other components more explicit.

Reply:
Added the `InputValidating` protocol with two validation methods. Updated `Validator` to conform to it, along with unit tests and the CLI example.